test/parallel/test-dns-lookup-promises-options-deprecated.js

```diff
@@ -0,0 +1,32 @@
+// Flags: --expose-internals
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { internalBinding } = require('internal/test/binding');
+const cares = internalBinding('cares_wrap');
+cares.getaddrinfo = () => internalBinding('uv').UV_ENOMEM;
+
+// This test ensures that dns.lookup issue a DeprecationWarning
+// when invalid options type is given
+
+const dnsPromises = require('dns/promises')
+
+common.expectWarning({
+  'internal/test/binding': [
+    'These APIs are for internal testing only. Do not use them.',
+  ],
+  'DeprecationWarning': {
+    DEP0153: 'Type coercion of dns.lookup options is deprecated'
+  }
+});
+
+assert.throws(() => {
+  dnsPromises.lookup('127.0.0.1', { hints: '-1' });
```

Comment:
This test only check whether DeprecationWarning is emitted when `hints` is not a number.
Since the valid string value of hints depends on the OS environment, give an invalid string value here and assert the error.